sdk/src/transaction.rs

```diff
@@ -86,6 +87,18 @@ pub struct Transaction {
     pub message: Message,
 }
 
+impl Sanitize for Transaction {
+    fn sanitize(&self) -> std::result::Result<(), SanitizeError> {
+        if self.message.header.num_required_signatures as usize > self.signatures.len() {
+            return Err(SanitizeError::Failed);
+        }
+        if self.signatures.len() > self.message.account_keys.len() {
```

Comment:
This looks like a duplicate check given the check above and the implementation of `sanitize` for `Instruction`

Reply:
we dont have access to signatures.len() in message. id rather have this check as well, instead of only relying on num_required_signatures